File: app/src/test/java/io/github/qwqgong/androidcyaml/ProcessMatchingModeTest.java
package io.github.qwqgong.androidcyaml;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertThrows;

import org.junit.Test;

public final class ProcessMatchingModeTest {
    @Test
    public void missingValuePreservesLegacyEnabledDefault() {
        assertEquals(ProcessMatchingMode.ALWAYS, ProcessMatchingMode.fromWireValue(null));
        assertEquals(ProcessMatchingMode.ALWAYS, ProcessMatchingMode.fromWireValue(""));
    }

    @Test
    public void parsesEveryMihomoMode() {
        for (ProcessMatchingMode mode : ProcessMatchingMode.values()) {
            assertEquals(mode, ProcessMatchingMode.fromWireValue(mode.wireValue()));
        }
        assertEquals(ProcessMatchingMode.STRICT, ProcessMatchingMode.fromWireValue(" STRICT "));
    }

    @Test
    public void rejectsUnknownMode() {
        assertThrows(
                IllegalArgumentException.class,
                () -> ProcessMatchingMode.fromWireValue("enabled")
        );
    }

    @Test
    public void runtimeDefaultsToAlways() {
        assertEquals(
                ProcessMatchingMode.ALWAYS,
                RuntimeOverrideSettings.defaults().processMatchingMode()
        );
    }
}

File: app/src/test/java/io/github/qwqgong/androidcyaml/ProcessMatchingModeTest.kt
package io.github.qwqgong.androidcyaml

import org.junit.Assert.assertEquals
import org.junit.Assert.assertThrows
import org.junit.Test

class ProcessMatchingModeTest {
    @Test
    fun missingValuePreservesLegacyEnabledDefault() {
        assertEquals(ProcessMatchingMode.ALWAYS, ProcessMatchingMode.fromWireValue(null))
        assertEquals(ProcessMatchingMode.ALWAYS, ProcessMatchingMode.fromWireValue(""))
    }

    @Test
    fun parsesEveryMihomoMode() {
        for (mode in ProcessMatchingMode.values()) {
            assertEquals(mode, ProcessMatchingMode.fromWireValue(mode.wireValue))
        }
        assertEquals(ProcessMatchingMode.STRICT, ProcessMatchingMode.fromWireValue(" STRICT "))
    }

    @Test
    fun rejectsUnknownMode() {
        assertThrows(IllegalArgumentException::class.java) {
            ProcessMatchingMode.fromWireValue("enabled")
        }
    }

    @Test
    fun runtimeDefaultsToAlways() {
        assertEquals(
            ProcessMatchingMode.ALWAYS,
            RuntimeOverrideSettings.defaults().processMatchingMode,
        )
    }
}
